src/Std/Data/HashMap/Basic.lean

```diff
@@ -0,0 +1,193 @@
+  getElem! m a := m.get! a
+
+@[inline, inherit_doc DHashMap.remove] def remove [BEq α] [Hashable α] (m : HashMap α β) (a : α) :
+    HashMap α β :=
+  ⟨m.inner.remove a⟩
+
+@[inline, inherit_doc DHashMap.size] def size [BEq α] [Hashable α] (m : HashMap α β) : Nat :=
+  m.inner.size
+
+@[inline, inherit_doc DHashMap.isEmpty] def isEmpty [BEq α] [Hashable α] (m : HashMap α β) : Bool :=
+  m.inner.isEmpty
+
+section Unverified
+
+/-! We currently do not provide lemmas for the functions below. -/
+
+@[inline, inherit_doc DHashMap.filter] def filter [BEq α] [Hashable α] (f : α → β → Bool)
+    (m : HashMap α β) : HashMap α β :=
+  ⟨m.inner.filter f⟩
+
+@[inline, inherit_doc DHashMap.foldlM] def foldlM [BEq α] [Hashable α] {m : Type w → Type w}
+    [Monad m] {γ : Type w} (f : γ → α → β → m γ) (init : γ) (b : HashMap α β) : m γ :=
+  b.inner.foldlM f init
+
+@[inline, inherit_doc DHashMap.foldl] def foldl [BEq α] [Hashable α] {γ : Type w}
+    (f : γ → α → β → γ) (init : γ) (b : HashMap α β) : γ :=
+  b.inner.foldl f init
+
+@[inline, inherit_doc DHashMap.forM] def forM [BEq α] [Hashable α] {m : Type w → Type w} [Monad m]
+    (f : (a : α) → β → m PUnit) (b : HashMap α β) : m PUnit :=
+  b.inner.forM f
+
+@[inline, inherit_doc DHashMap.forIn] def forIn [BEq α] [Hashable α] {m : Type w → Type w} [Monad m]
+    {γ : Type w} (f : (a : α) → β → γ → m (ForInStep γ)) (init : γ) (b : HashMap α β) : m γ :=
+  b.inner.forIn f init
+
+instance [BEq α] [Hashable α] {m : Type w → Type w} : ForM m (HashMap α β) (α × β) where
+  forM m f := m.forM (fun a b => f (a, b))
+
+instance [BEq α] [Hashable α] {m : Type w → Type w} : ForIn m (HashMap α β) (α × β) where
+  forIn m init f := m.forIn (fun a b acc => f (a, b) acc) init
+
+@[inline, inherit_doc DHashMap.Const.toList] def toList [BEq α] [Hashable α] (m : HashMap α β) :
+    List (α × β) :=
+  DHashMap.Const.toList m.inner
+
+@[inline, inherit_doc DHashMap.Const.toArray] def toArray [BEq α] [Hashable α] (m : HashMap α β) :
+    Array (α × β) :=
+  DHashMap.Const.toArray m.inner
+
+@[inline, inherit_doc DHashMap.keys] def keys [BEq α] [Hashable α] (m : HashMap α β) : List α :=
+  m.inner.keys
+
+@[inline, inherit_doc DHashMap.keysArray] def keysArray [BEq α] [Hashable α] (m : HashMap α β) :
+    Array α :=
+  m.inner.keysArray
+
+@[inline, inherit_doc DHashMap.values] def values [BEq α] [Hashable α] (m : HashMap α β) : List β :=
+  m.inner.values
+
+@[inline, inherit_doc DHashMap.Const.insertMany] def insertMany [BEq α] [Hashable α] {ρ : Type w}
+    [ForIn Id ρ (α × β)] (m : HashMap α β) (l : ρ) : HashMap α β :=
+  ⟨DHashMap.Const.insertMany m.inner l⟩
+
+@[inline, inherit_doc DHashMap.Const.insertManyUnit] def insertManyUnit [BEq α] [Hashable α]
+    {ρ : Type w} [ForIn Id ρ α] (m : HashMap α Unit) (l : ρ) : HashMap α Unit :=
+  ⟨DHashMap.Const.insertManyUnit m.inner l⟩
+
+@[inline, inherit_doc DHashMap.Const.ofList] def ofList [BEq α] [Hashable α] {ρ : Type w}
+    [ForIn Id ρ (α × β)] (l : ρ) : HashMap α β :=
```

Comment:
This type signature also seems to not match the name (see comment on underlying functions)